pyam/core.py

```diff
@@ -1382,6 +1382,58 @@ def map_regions(self, map_col, agg=None, copy_col=None, fname=None,
         if not inplace:
             return ret
 
+    def subtract(self, other, join_col, new_name):
+        """
+        Subtract data in ``other`` from ``self``
+
+        Parameters
+        ----------
+        other : pyam.IamDataFrame
+            Object containing data to subtract
+
+        join_col : str
+            Column to use to subtract the two sets of data (e.g. ``variable``)
+
+        new_name : str
+            Name to assign to ``join_col`` in the output timeseries
+
+        Raises
+        ------
+        ValueError
+            The metadata columns in ``self`` and ``other`` are not identical
+
+        ValueError
+            ``self`` or ``other`` contains more than value for ``join_col``
+
+        NotImplementedError
+            The type of ``other`` is not yet supported
+        """
+        if not isinstance(other, IamDataFrame):
+            raise NotImplementedError
+
+        too_many_vals_error = "`{}` contains more than one entry for `{}`"
+        if len(self[join_col].unique()) > 1:
+            raise ValueError(too_many_vals_error.format("self", join_col))
+
+        if len(other[join_col].unique()) > 1:
+            raise ValueError(too_many_vals_error.format("other", join_col))
+
+        s_ts = self.timeseries()
+        o_ts = other.timeseries()
+
+        if set(s_ts.index.names) != set(o_ts.index.names):
```

Comment:
I'd suggest to move the `meta`-equality assertion from `append()` to an own auxiliary function and re-use here, https://github.com/IAMconsortium/pyam/blob/a4e266c93f531f94285311fd354aaa881aa403c4/pyam/core.py#L202

Reply:
I don't really understand what the meta checks in `append` are doing so have just taken the line you linked. Happy to refactor further if you can see how to do it.

Reply:
Yeah, I meant the entire section of the code, not just one line. I implemented the change for you, see https://github.com/znicholls/pyam/pull/3 - should be pretty straightforward to re-use it now in `subtract()`.

Reply:
Wicked thanks mate! Have pulled in and used now. The next question is what should happen when you're subtracting with the `join_col` being `model` or `scenario`. I guess you just get a brand new, empty `meta` table as you're resulting `model` or `scenario` won't exist in the `meta` table you started with?

Reply:
Yeah, good question... I'd suggest to either:
 - Don't allow subtracting on `model` or `scenario` for this PR and tackle later.
 - Start with an empty `meta` table

Reply:
I've implemented empty (after all, if you subtract scenarios, it's more or less a new scenario right?)